### DISCUSSION TITLE
New Dexcom 8T series G6 transmitter refuses to display data on Xdrip+ but works on Dexcom receiver

### DISCUSSION THREAD

**Discussion by bwschulz** [Navid200/xDrip#72] [Dexcom]
Hey everyone!
First time posting because I've got an odd Xdrip+ problem that has been unsolvable via my usual resources, so I'm reaching out to the experts here.
I just had to switch Dexcom G6 transmitters and my new one (series 8T) is refusing to display data on Xdrip+.
I have restarted my phone and updated my Xdrip+ install to the latest nightly release (9/2/21) and still nothing.
There may also be an issue with the transmitter code. Every transmitter I can remember used the following numbering convention, where @ are letters and # are numbers: 8@##@@
But the code for this transmitter ends in a 0 (zero). Not an O (capital o). The transmitter had not been connecting and I was trying all kinds of different things to make it work. Changing what I though was the last letter for the transmitter to a zero is what finally did it.
But now that the transmitter is connected I get no readings in Xdrip+, although it connected to and gives reading to the Dexcom receiver just fine. This behavior has persisted through the initial restarted sensor and now to a new one. Here are the error codes and details:

Main Screen:
"G6 State isn't currently known. Next connection will update this"

Classic Status Page
Bluetooth Device: None Set
Connection Status: Not Connected
(But the sensor is started and running)

G6/G6 Status Page
Bluetooth Link: Disconnected, Connecting, or Connected (screenshot below shows while connected)
Last Connected: Up to several minutes ago (it is connecting to something)
Brain State: Scanning Errors: 217 (when not connected) or Bonding Errors: 218 (when connected)

Screenshots of status page- https://photos.app.goo.gl/7x9DnuGQU4Np9Gw7A 

Here are some selected lines from the Event Log:

DexTimeKeeper
[timestamp] No valid timestamp stored for transmitter: 8T##@0

Ob1G5StateMachine
[timestamp] Unhandled packet type in reply: InvalidRxMessage
FF0601

Ob1G5Collection Service
[timestamp] Error count reached: [increasing four digits]

Ob1G5StateMachine
[timestamp] authentication notification throwable: (CONNECT_NOW)
com.polidea.rxandroidable2.exceptions.BleDisconnectedException:
Disconnected from MAC=‘XX:XX:XX:XX:XX:XX’ with status 0 (GATT_SUCCESS)
[date & time]

Please help if you can or let me know where else I should look!

**Comment by Navid200** [comment]
![Screenshot_20211008-161117](https://user-images.githubusercontent.com/51497406/136622043-d48aab64-e853-446d-937f-bb7287bf3ac9.jpg)

According to this system status page, there is no connectivity.  Hence, the items stuck in the queue.
As a first step, you should clear the queue:
https://github.com/Navid200/xDrip/wiki/Clear-Queue

**Reply by bwschulz** [reply]
Thanks for the suggestion. I cleared it as instructed, but still have the problem. I was entering calibrations, carbs, and insulin to keep a time record of them (and hoping that it would suddenly start working) but I will stop and not enter any more if that could be part of the problem.

**Reply by Navid200** [reply]
I said, as a first step

**Comment by Navid200** [comment]
Now, check all the settings:
https://github.com/Navid200/xDrip/wiki/G6-Recommended-Settings

**Reply by bwschulz** [reply]
It's been a while since I checked these and I did need to tick "Avoid Scanning" and "Authenticate G5 before each read".
I also needed to enabled data usage in Android settings.
After restarting I tried messing around with the transmitter ID again (described in response to emp-00 below) and this time it worked. So thanks to both of you- I don't know what exactly made it work again, but it is and that's what matters!

**Comment by emp-00** [comment]
Maybe better continue here, just pasting my answers:
"Bonding Error" - I've never seen/never had this ... very, very odd. Sorry ... looks maybe like a bluetooth connection issue? Have you tried restarting your phone? I would try => "Forget Device" and re-pair your transmitter after restarting the phone. Make sure you have 100% sure the correct transmitter ID (you mentioned an issue with "O" and "0" ... if you mix this up you cannot connect)

**Reply by bwschulz** [reply]
I've done the restart and forget device things a few times. I'm sure that the transmitter ID is correct because that was the problem before this one. It definitely has a connection because the Dexcom receiver that is using the same ID is working perfectly. G5G/6 status says Bluetooth Link: Connected, while the system status says Bluetooth Device: None Set and Connection Status: Not Connected.

But I did manage to fix it; possibly involving forget device. Even though I've already tried this before and it didn't work, I changed the transmitter ID back to the wrong one, forgot device, and restarted collector. Then I changed the transmitter ID back to the correct one, forgot device, restarted collector (I think I also toggled BT on and off at some point as well). Then in a few minutes BAM. Found device- data coming in. Yay!

**Comment by Navid200** [comment]
Those are all symptoms, not the issue.  The issue is there is no proper connectivity.  I am going through all the required steps to fix it.

**Reply by bwschulz** [reply]
It's fixed now. After I cleared the queue, fixed those settings, and restarted I tried messing with the transmitter ID and forgetting the device again (which didn't work the other times I had tried it) but this time it connected. So thanks for the help!

**Reply by Navid200** [reply]
That's great
